telosnetwork/telos-wallet
# Login with Social Media

# Fixes #463

## Description
This PR includes a new button for the Home page (on the EVM side) which provides the users a way to authenticate using social media providers without even having a wallet installed.

## Notes
- I removed the timeout when sending tokens because it was annoying. Maybe we should discuss if we want this timeout or not.
- I'm currently using [my own app id for testing because I needed to do some tests directly](https://github.com/telosnetwork/telos-wallet/issues/492#issuecomment-1646256233). We need to decide which Identity providers are we going to be using and enable them in the app registration.

## Test scenarios
- go to [this link](https://deploy-preview-490--wallet-staging.netlify.app/)
- Press the "Login with Social Media" button
   - A menu with 4 social media authentication providers should appear
- Choose any provider and press the button
   - a window will pop up to ask you for a pin (If is your first time, it will ask you to repeat the pin)
   - after confirmation
     - the window will load and close automatically
     - you will be logged in with a zero balance account
- Copy your address, and send to that address some tokens from another account you control
- After you have some tokens in your balance, go to send page and try to send them (make sure you have some TLOS)
- When pressing Send button, a window will pop up
- Enter PIN and confirm
  - the transaction should go with no problems.



<table>
<tr>
<td><img src="https://github.com/telosnetwork/telos-wallet/assets/4420760/33916774-f4b8-4aec-af51-a21621d7bf1d"></td>
<td><img src="https://github.com/telosnetwork/telos-wallet/assets/4420760/642f42b8-d2bc-487e-8d96-9e34ef13972c"></td>
</tr>
</table>


## Checklist:
